src/ixdat/backends/directory_backend.py

```diff
@@ -0,0 +1,158 @@
+
+
+class DirBackend(BackendBase):
+    """A database backend that loads and saves .ix files from a directory"""
+
+    def __init__(
+        self,
+        directory=STANDARD_DATA_DIRECTORY,
+        suffix=STANDARD_SUFFIX,
+        data_suffix=STANDARD_DATA_SUFFIX,
+    ):
+        """Initialize a directory database backend with the directory as Path
+
+        Args:
+            directory (Path): the directory to save files (under table subfolders)
+            suffix (str): The suffix to use for JSON-formatted metadata files
+            data_suffix (str): The suffix to use for numpy-formatted data files
+        """
+        self.directory = directory
+        self.suffix = suffix
+        self.data_suffix = data_suffix
+
+    @property
+    def name(self):
+        return f"DirBackend({self.directory})"
+
+    def save(self, obj):
+        """Save the Saveable object as a file corresponding to a row in a table"""
+        if obj.data_objects:
+            # save any data objects first as this may change the references
+            for data_obj in obj.data_objects:
+                self.save_data_obj(data_obj)
+        try:
+            table_name = obj.table_name
+        except AttributeError:
+            table_name = str(type(obj))
+        obj_as_dict = obj.as_dict()
+        obj.id = self.add_row(obj_as_dict, table_name=table_name)
+        obj.backend_name = self.name
+        return obj.id
+
+    def open(self, cls, i):
+        """Open a Saveable object represented as row i of table cls.table_name"""
+        table_name = cls.table_name
+        obj_as_dict = self.open_serialization(table_name, i)
+        obj = cls.from_dict(obj_as_dict)
+        obj.backend = self
+        return obj
+
+    def contains(self, table_name, i):
+        """Check if id `i` is already a principle key in the table named `table_name`"""
+        return i in self.get_id_list(table_name)
+
+    def load_obj_data(self, obj):
+        """Return the data for an object loaded from its .ixdata file"""
+        if not hasattr(obj, "data"):
+            # there's no data to be got or obj already has its data
+            return
+        path_to_row = self.get_path_to_row(obj.table_name, obj.id)
+        try:
+            return np.load(path_to_row.with_suffix(self.data_suffix))
+        except FileNotFoundError:
+            # there's no data to be got.
+            return
+
+    def save_data_obj(self, data_obj):
+        """Save the object as a .ix for metadata and .ixdata for numerical data"""
+        table_name = data_obj.table_name
+        if data_obj.backend == self and self.contains(table_name, data_obj.id):
+            return data_obj.id  # already saved!
+        obj_as_dict = data_obj.as_dict()
+        data = obj_as_dict["data"]
+        obj_as_dict["data"] = None
+        data_obj.id = self.add_row(obj_as_dict, table_name=table_name)
+        folder = self.directory / table_name
+        data_obj.backend = self
```

Comment:
oh, no. They should be the same. I guess saving `backend` is better than saving` backend_name`? I will make it so, please say if you think otherwise. The only place it's used later is when saving, an object checks the backend to see if its already saved...

Reply:
Agreed better to save backend